protocol/nats_jetstream/v2/sender.go

```diff
@@ -90,11 +90,23 @@ func (s *Sender) Send(ctx context.Context, in binding.Message, transformers ...b
 		}
 	}()
 
+	ctxHeader := context.WithValue(ctx, natsHeaderContextKey, natsHeader{})
```

Comment:
Is there any use for an empty `natsHeader` here?

Reply:
The code does not work properly without passing in an empty object.  I debated changing the signature of WriteMsg(...) to return a header and error.  That way there is no need for the context.  Let me know which way you would like me to proceed.

Reply:
IIUC, you control `WriteMsg(...)` no? So you can change as needed.

Reply:
I simplified the code to not use the context to pass information.   It now returns the nats.Header out of the function.